### PR TITLE
Various fixes & cleanup

### PR DESCRIPTION
**Pull request by Herrie82**
- Disable QML role file generation
- Fix service registration and permissions
- Cleanup of code no longer needed

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

File: Src/remote/ApplicationProcessManager.cpp [hunk 1]
```diff
@@ -424,7 +424,7 @@ qint64 ApplicationProcessManager::launchQMLApp(ApplicationDescription *desc, std
     if (appParams.length() > 0)
 		parameters << appParams;
 
-	qmlGenerateRole(desc->id());
+	//qmlGenerateRole(desc->id());
```
Comment (Tofee) [inline Src/remote/ApplicationProcessManager.cpp:427]: I understand that librolegen (which implements qmlGenerateRole, with old pub/priv roles) is now a bit useless, but why did it need to generate roles on the fly here ?

Comment (Herrie82) [reply Src/remote/ApplicationProcessManager.cpp:427]: I really don't know... Ask morphis... His commit messages weren't usually very detailed:

https://github.com/webOS-ports/librolegen/commit/afaab12fc9786a1b1f39998ac9ea1d96364469f7
https://github.com/webOS-ports/luna-appmanager/commit/f1bcfac9f041c213a69eaf758e7b34a87074c262
https://github.com/webOS-ports/luna-appmanager/commit/76bae5aa2e5871a02c6ced0cb502bde843648446

Comment (Tofee) [reply Src/remote/ApplicationProcessManager.cpp:427]: I really don't see any usecase for this. Nowadays the roles are shipped with each app, and we start all apps via SAM... "launchQMLApp" is probably useless now.

Comment (Tofee) [reply Src/remote/ApplicationProcessManager.cpp:427]: I've looked around a bit, and it looks like we start an app "the old way" when the SAM call has failed, see https://github.com/webOS-ports/luna-appmanager/blob/master/Src/remote/ApplicationProcessManager.cpp#L192

Comment (Tofee) [reply Src/remote/ApplicationProcessManager.cpp:427]: But do we actually need a fallback, I'm not so sure.

Comment (Herrie82) [reply Src/remote/ApplicationProcessManager.cpp:427]: I am also not sure.... SAM should be able to launch everything, shouldn't it?

Comment (Tofee) [reply Src/remote/ApplicationProcessManager.cpp:427]: Yes, it should. I'm quite in favor of just dropping the whole fallback thing. I'll try and do that in another PR, yours here is fine as it is.

Comment (Herrie82) [reply Src/remote/ApplicationProcessManager.cpp:427]: I'll do a bit of cleanup. Consider this WIP for now

**Comment by Herrie82** [comment]
@Tofee Cleaned up some bits. Seems to behave on qemux86-64 here.

File: Src/remote/ApplicationProcessManager.cpp [hunk 2]
```diff
@@ -123,76 +89,71 @@ QList<ApplicationInfo*> ApplicationProcessManager::runningApplications() const
...
+    if(params.empty()) params = "{}";
+    std::string SAM_params = "{ \"id\": \"" + appId + "\", \"params\": " + params + " }";
+    g_warning("Delegating launch call to SAM...");
+    if (LSCall(ApplicationManager::instance()->getServiceHandle(),
+                "luna://com.webos.service.applicationManager/launch", SAM_params.c_str(),
+                NULL, NULL, NULL, &lserror))
+    {
+        return std::string("");
+    }
+
```
[the remark this answers is not in the document]
Comment (Herrie82) [reply Src/remote/ApplicationProcessManager.cpp]: @Tofee like this? https://github.com/webOS-ports/luna-appmanager/pull/41/commits/998053e5c20ca35eab1f6de74a9f0be5e9a44993

[the remark this answers is not in the document]
Comment (Tofee) [reply Src/remote/ApplicationProcessManager.cpp]: perfect !

**Pull request merged**